src/concrete/ml/sklearn/base.py

```diff
@@ -657,7 +657,10 @@ def predict(self, X: Data, fhe: Union[FheMode, str] = FheMode.DISABLE) -> numpy.
                 # Execute the inference in FHE or with simulation
                 q_y_pred_i = predict_method(q_X_i)
 
-                q_y_pred_list.append(q_y_pred_i[0])
+                if self.__class__.__name__ == "KNeighborsClassifier":
+                    q_y_pred_list.append(q_y_pred_i)
+                else:
+                    q_y_pred_list.append(q_y_pred_i[0])
```

Comment:
this does not look right, the output of a circuit is a tuple of one element (since we only pass batches of 1 as inputs), this is why we have this `[0]`. I can't see why a specific model should do that differently

Reply:
@kcelia

Reply:
I'm addressing all your comments.
I said in https://zama-ai.slack.com/archives/C0455QDGT6C/p1694684283172319?thread_ts=1694590559.480319&cid=C0455QDGT6C, I agree with you.
I was a workaround.

Reply:
ah yes I'm not following this discussion ! thanks

Reply:
fixed